wdn/templates_4.1/scripts/search.js

```diff
@@ -99,29 +106,78 @@ define(['jquery', 'wdn', 'require', 'modernizr', 'navigation'], function($, WDN,
 				});
 				domSearchForm.append(domEmbed);
 
+				var createSearchFrame = function() {
+					// lazy create the search iframe
+					if (!$unlSearch) {
+						$unlSearch = $('<iframe>', {
+							name: 'unlsearch',
+							id: 'wdn_search_frame',
+							title: 'Search results',
+							src: searchFrameAction
+						});
+
+						domSearchForm.parent().append($unlSearch).append($progress);
+
+						$unlSearch.on('load', function() {
+							postReady = true; // iframe should be ready to post messages to
+						});
+					}
+				};
+
+				var activateSearch = function() {
+					domSearchForm.parent().addClass('active');
+					$progress.show();
+				};
+
+				var postSearchMessage = function(query) {
+					$unlSearch[0].contentWindow.postMessage(query, searchOrigin);
+					$progress.hide();
+				};
+
+				var closeSearch = function() {
+					domSearchForm.parent().removeClass('active');
+					domSearchForm[0].reset();
+				};
+
 				// add an event listener to support the iframe rendering
-				domQ.on('keyup', function(e0) {
+				domQ.on('keyup', function(e) {
 					// ONLY for "desktop" presentation
 					if (!isFullNav()) {
 						return;
 					}
 
-					if (e0.keyCode === 27) {
+					var keyCode = e.keyCode;
+
+					if (keyCode === 27) {
 						//Close on escape
 						closeSearch();
 						return;
 					}
 
+					// ignore non-printable keys (blacklist)
+					if ((keyCode !== 32 && keyCode < 48) ||
+						(keyCode > 90 && keyCode < 96) ||
+						(keyCode > 111 && keyCode < 186 && keyCode !== 173) ||
+						(keyCode > 192 && keyCode < 219) ||
+						(keyCode > 222)
+					) {
+						return;
+					}
+
 					clearTimeout(autoSubmitTimeout);
+
 					if ($(this).val()) {
+						// activate search UI
+						createSearchFrame();
+						activateSearch();
+
+						// debounce auto-submit
 						autoSubmitTimeout = setTimeout(function() {
 							domSearchForm.trigger('submit', 'auto');
-						}, 300);
+						}, autoSearchDebounceDelay);
```

Comment:
You are correct, that is currently the expectation in the code. However, the longer timeout makes the issue noticeable.